check-submodule.py

```diff
@@ -158,8 +183,14 @@ def dump_minimal_tests(
     filename = code_spans_object["name"]
 
     # Read code spans from the input source code
-    with open(filename, "r") as input_file:
-        lines = input_file.readlines()
+    with open(filename, "r", encoding="utf-8") as input_file:
```

Comment:
Nit: You could ignore encoding errors and run on those files anyway

Reply:
encoding="utf-8" is the default encoding mode used by `Python`, but to avoid problems with the different versions, I stated it explicitly. Or are you advising to convert non-utf-8 files into utf-8?

Reply:
Yep, I know, I was suggesting ignoring encoding errors.
I'm sure you can do that by reading bytes and then `decode('utf-8', 'ignore')`, maybe there is a way to do the same with `open`.